### PR TITLE
repo: Dev 1.20.8 + refresh test certs 

### PR DESCRIPTION
**Pull request by oschaaf**
- Kick off v1.20.8
- Refresh integration test certificates (some expired) 

Signed-off-by: Otto van der Schaaf <ovanders@redhat.com>

### CODE REVIEW

**Comment by oschaaf** [comment]
Whoops, looks like running `certs.sh` in `test/config/integration/certs` wasn't enough:

```bash
➜  ~ curl -s https://raw.githubusercontent.com/envoyproxy/envoy/release/v1.20/test/extensions/transport_sockets/tls/test_data/ca_cert.pem \ 
| openssl x509 -noout -enddate
notAfter=Aug 20 16:57:46 2022 GMT
```

It looks like running `test/extensions/transport_sockets/tls/test_data/certs.sh` and committing the updated certificates will solve the remaining test failures.

**Comment by oschaaf** [comment]
Allright, this is now down to `TestSPIFFEValidator.TestDaysUntilFirstCertExpires` failing at https://github.com/envoyproxy/envoy/blob/release/v1.20/test/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator_test.cc#L529. 

Looking at that test, it harcodes an assumed number of days till expiry. That won't work in conjunction with refreshing the certificate involved:

```
2022-09-01T07:19:17.5041036Z [ RUN      ] TestSPIFFEValidator.TestDaysUntilFirstCertExpires
2022-09-01T07:19:17.5041910Z TestRandomGenerator running with seed -171874280
2022-09-01T07:19:17.5042668Z test/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator_test.cc:529: Failure
2022-09-01T07:19:17.5043356Z Expected equality of these values:
2022-09-01T07:19:17.5043829Z   19231
2022-09-01T07:19:17.5044304Z   validator().daysUntilFirstCertExpires()
2022-09-01T07:19:17.5044817Z     Which is: 19966
2022-09-01T07:19:17.5045266Z Stack trace:
2022-09-01T07:19:17.5046143Z   0x6628d3: Envoy::Extensions::TransportSockets::Tls::TestSPIFFEValidator_TestDaysUntilFirstCertExpires_Test::TestBody()
2022-09-01T07:19:17.5047076Z   0xfd2228: testing::internal::HandleExceptionsInMethodIfSupported<>()
2022-09-01T07:19:17.5047720Z   0xfd2155: testing::Test::Run()
2022-09-01T07:19:17.5048300Z   0xfd2ce0: testing::TestInfo::Run()
2022-09-01T07:19:17.5048855Z ... Google Test internal frames ...
2022-09-01T07:19:17.5049172Z 
2022-09-01T07:19:17.5049766Z test/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator_test.cc:531: Failure
2022-09-01T07:19:17.5050467Z Expected equality of these values:
2022-09-01T07:19:17.5050940Z   19221
2022-09-01T07:19:17.5051423Z   validator().daysUntilFirstCertExpires()
2022-09-01T07:19:17.5051934Z     Which is: 19956
2022-09-01T07:19:17.5052362Z Stack trace:
2022-09-01T07:19:17.5053089Z   0x6629e2: Envoy::Extensions::TransportSockets::Tls::TestSPIFFEValidator_TestDaysUntilFirstCertExpires_Test::TestBody()
2022-09-01T07:19:17.5054127Z   0xfd2228: testing::internal::HandleExceptionsInMethodIfSupported<>()
2022-09-01T07:19:17.5054766Z   0xfd2155: testing::Test::Run()
2022-09-01T07:19:17.5055316Z   0xfd2ce0: testing::TestInfo::Run()
2022-09-01T07:19:17.5055863Z ... Google Test internal frames ...
2022-09-01T07:19:17.5056162Z 
2022-09-01T07:19:17.5056686Z [  FAILED  ] TestSPIFFEValidator.TestDaysUntilFirstCertExpires (10 ms)
```

I propose we just update the expectations; with that we should at least be good until 1.20 slides out of the support timeframe.

**Comment by oschaaf** [comment]
Hmm Ci failed because the `linux_x64 compile_time_options` task timed out after executing for two hours; I can't really do anything about that. Other then that, I think these changes passed CI tests, so I'll await review feedback (and not attempt to resolve `linux_x64 compile_time_options`).

File: docs/root/version_history/v1.20.7.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+Minor Behavior Changes
+----------------------
+*Changes that may cause incompatibilities for some users, but should not for most*
+
+Bug Fixes
+---------
+*Changes expected to improve the state of the world and are unlikely to have negative effects*
+
+* ci: fix disk space issue that have prevented publication.
```
Comment (phlax) [inline docs/root/version_history/v1.20.7.rst:16]: this doesnt seem to match the changelog that it was copied from

**Review by phlax — COMMENTED** [review]
thanks for picking this up @oschaaf - small nit inline

File: docs/root/version_history/v1.20.7.rst (hunk 1 above)
Comment (oschaaf) [reply docs/root/version_history/v1.20.7.rst:16]: Thank you for catching that. I was tired and it seems I copied https://raw.githubusercontent.com/envoyproxy/envoy/release/v1.20/docs/root/version_history/v1.20.6.rst where I ought to have copied https://github.com/envoyproxy/envoy/blob/release/v1.20/docs/root/version_history/current.rst

I will amend & force push so we keep two commits.

Comment (oschaaf) [reply docs/root/version_history/v1.20.7.rst]: So I pushed the change to amend this, and as a bonus CI turned green 🥳 
Ready for another pass

**Review by phlax — APPROVED** [review]
great! Thanks @oschaaf for resolving this

**Comment by oschaaf** [comment]
Thank you for the review @phlax !

**Pull request merged**